### PR TITLE
Integrate website into the list

### PR DESCRIPTION
**Pull request by Manvel**
# tl;dr

Took some time to integrate website into the listing, as the result now the [website is alive on Github Pages](https://manvel.github.io/staticsitegenerators-list/) with the full updated list.

# Background

Currently the website is outdated, I've added a SSG [here](https://github.com/bevry/staticsitegenerators-list/pull/322) and fixed the errors [here](https://github.com/bevry/staticsitegenerators-list/pull/323), but yet [https://staticsitegenerators.net/](https://staticsitegenerators.net/) is not updated.

@balupton noted [here](https://github.com/bevry/staticsitegenerators-list/issues/321) and [here](https://github.com/bevry/staticsitegenerators-website/issues/29) that the dual repo setup is the main reason for the issue.

## Integration notes

- Moved the website into `/website`
- `/website/cmints.js` is the integration file, source of magic.
- `/website/config.js` - the configuration file, contains same website configuration data from [docpad.js](https://github.com/bevry/staticsitegenerators-website/blob/master/docpad.js).
- `/website/theme/layouts` - contains layouts and partials(leading "_" in the filename). This is similar to the [Original layouts](https://github.com/bevry/staticsitegenerators-website/tree/master/src/layouts), with couple of differences:
  - Using [.ejs](http://ejs.co/) instead of [eco](https://github.com/sstephenson/eco).
  - As a result using partials instead of inheritance.
- `/website/pages` - contain pages, similar to the pages in [render folder](https://github.com/bevry/staticsitegenerators-website/tree/master/src/render), with some differences:
  - `.ejs` and `.html` is used instead of `.eco`
  - `list.json` is copied from the project root directory "on start" instead of `.json` generation as in the [original](https://github.com/bevry/staticsitegenerators-website/blob/master/src/render/list.json.eco) project.
- `/website/layout/less` - where the main styles [from original](https://github.com/Manvel/staticsitegenerators-list/blob/create-website/website/cmints.js#L8) project resides, the difference is that it's using `.less` instead of `.eco`.
- `/website/public` - same as [raw in the original project](https://github.com/bevry/staticsitegenerators-website/tree/master/src/raw).
- `/website/content` - This is the directory where the project is built

## Additional changes in current PR

- Generated new [Google Plus button](https://developers.google.com/+/web/share/), as the old one were throwing error in my local environment. The difference is that it currently it also contain word "share" next to the logo.
- Fixed the social media buttons issue, that they are currently broken and most of them below the fold on my Chrome v69.
- Added missing Bitbucket repo links.
- Changed resources to use relative URLs instead of root URLs, this was only done because of the Github Pages, because otherwise resources would have been unreachable on [Github Pages](https://manvel.github.io/staticsitegenerators-list/).
- Fixed additional issues in `list.json` causing the build to be broken, mostly was again http to https changes in the websites.
- Added note in the README documentation as it wasn't obvious how to use Github credentials.

## Adjustments planned for future

There are several things that can be improved IMO:
- Isolate the Social widgets in separate iframe sandboxes, because having 1 big iframe sandbox makes it hard to adjust and have potential to lead again to the issue of the buttons moving below the fold.
- Table adjustments:
  - Lazy load the table
  - Improve table accessibility
  - Make table Responsive
- Make the website more appealing
- Add filters to search in the list
- Introduce dedicated page for Static Site Generators
- Localization and global SEO 

## Usage

**Note:** In order to avoide `API rate limit exceeded` error, setup `process.env.GITHUB_CLIENT_ID` and `process.env.GITHUB_CLIENT_SECRET` accordingly.

### Launching server

Command below runs tests, and server

```
npm start
```

### Publishing changes

Command below runs tests, generates static content and publish changes to the
github pages

```
npm run publish:gh-pages
```

## Hosting

I don't know where the pages currently are hosted, is there a reason why not in the GIthub?

Currently CMintS does generate both `.html` and `.gzip` for the content, for the Github I have removed generated `.gzip` files manually, but it's quite easy to automate that, or if needed I can introduce a new flag, for connivence.

## Resources

Result -> https://manvel.github.io/staticsitegenerators-list/
CMintS -> https://cmints.io/

### CODE REVIEW

**Comment by Manvel** [comment]
@balupton there is a bank holiday today(Wednesday) in Germany, so I'll be happy to address all the required changes you wish me to fix in current review. Otherwise I might catch up over the weekend, if you will have a chance to have a look.

**Comment by balupton** [comment]
> @balupton there is a bank holiday today(Wednesday) in Germany, so I'll be happy to address all the required changes you wish me to fix in current review. Otherwise I might catch up over the weekend(if you will have a chance to review).

Some deadlines sprang on me, so this got queued for later. Will get to it, but gotta get to some time urgent tasks first.

**Comment by dertuxmalwieder** [comment]
Hmm - Bitbucket repo links are *still* not in the list? (E.g. blogc++)

**Comment by Manvel** [comment]
> Hmm - Bitbucket repo links are still not in the list? (E.g. blogc++)

I can't see them being in the list on [https://staticsitegenerators.net/](https://staticsitegenerators.net/) as well, but should be simple to fix. I can fix that in the current review if @dertuxmalwieder and/or @balupton agree.

**Comment by dertuxmalwieder** [comment]
It would make sense, since we have had them in the list file for eternities.

**Comment by Manvel** [comment]
@dertuxmalwieder Ready, they should be available in the [Github Page](https://manvel.github.io/staticsitegenerators-list/). (CC: @balupton)

**Comment by dertuxmalwieder** [comment]
Confirmed, very nice, thank you!

File: list.json [hunk 1]
```diff
@@ -499,13 +499,6 @@
     "language": "Fortran",
     "licence": "GPL-3.0"
   },
-  {
-    "name": "Firedrop2",
-    "website": "http://www.voidspace.org.uk/python/firedrop2/",
```
Comment (Manvel) [inline list.json]: Current website doesn't seem to exist anymore, causing the tests to fail.

File: list.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "name": "ABlog",
-    "github": "abakan/ablog",
+    "github": "abakan-zz/ablog",
```
Comment (Manvel) [inline list.json]: Repository in moved to the new location.

File: list.json (hunk 1 above)
Comment (dertuxmalwieder) [reply list.json]: But why would you remove the whole SSG then?

**Comment by Manvel** [comment]
The PR has been updated, in order to push the changes to the Github pages it's enough to run:

```
npm run publish:gh-pages
```

@balupton Any hope you might have a chance to review the changes ?

(CC: @dertuxmalwieder )

**Comment by dertuxmalwieder** [comment]
Most of them look OK to me, except the one on which I commented above.

File: list.json (hunk 1 above)
Comment (Manvel) [reply list.json]: Seems like either website or repository link are required.

If I only remove the website property error below is thrown:

```
Comparison Error:
AssertionError [ERR_ASSERTION]: missing required fields on Firedrop2
    at equal (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/assert-helpers/source/index.js:171:10)
    at /Users/manvelsaroyan/projects/static-gen-list/master/source/test.js:59:4
    at Array.forEach (<anonymous>)
    at EventEmitterGrouped.<anonymous> (/Users/manvelsaroyan/projects/static-gen-list/master/source/test.js:56:8)
    at ambi (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/ambi/source/index.js:91:18)
    at Domain.fireMethod (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/taskgroup/source/lib/task.js:519:5)
    at Domain.run (domain.js:242:14)
    at EventEmitterGrouped.fire (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/taskgroup/source/lib/task.js:537:15)
    at Immediate.queue (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/taskgroup/source/lib/task.js:572:20)
    at runCallback (timers.js:810:20)
```

If I keep the non existing website test are failing:

```
Error #1:
static site generators list ➞ uris are valid / still exist ➞ Firedrop2: http get website: http://www.voidspace.org.uk/python/firedrop2/
AssertionError [ERR_ASSERTION]: response http status code should be 200 success
    at equal (/Users/manvelsaroyan/projects/static-gen-list/master/node_modules/assert-helpers/source/index.js:171:10)
    at /Users/manvelsaroyan/projects/static-gen-list/master/source/test.js:27:4
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

Googling didn't help me to find any existing repository associated with current project.

Having unmaintained list item causing the list being outdated, sounded to me something that's easier to remove.

**Comment by balupton** [comment]
I've taken the time to review this.

---

Regarding the listing repo.

With regards to the latest ecosystem situation, I think having the repos seperate is a better idea, but just dramatically revising how we do it — currently it uses travis ci and github hooks to trigger a build in the other repo, which is very fragile as it depends on up to date keys.

I will update this repo so that after successful builds, it will bump the patch version and do a new release of the package. This will solve the need to fetch a remote listing, as the package publish will always be the latest.

Will do this right now.

---

Regarding the website repo.

I don't really want us to tie it into any particular solution. As that could show preference of one solution over another.

I would be happy to setup cmints.staticsitegenerators.net as well as docpad.staticsitegenerators.net or nextjs.staticsitegenerators.net etc etc. And we can just redirect the homepage to whatever is most maintained.

This follows the whole idea of two repos, with the listing being a data dump hat could be implemented by whatever rendering sites they want — true jamstack stile.

---

Regarding your efforts.

You have done a terrific job considering the lack of information about this repo and its maintenance. I wish you reached out before you started your conversion, as we could have worked to something more inline with the vision I have for it.

You have done good work, and you should be proud.

Let's just try and figure out a way that incorporates this feedback.

---

For now, I will close this. Let me push up the new package publishing script, then we can take a stab at what to do about the website.

**Comment by balupton** [comment]
Hit the rate limits. New release will go live in an hour or so.

**Comment by Manvel** [comment]
@balupton Thanks for the feedback,

Looking forward to see the release going out .

I completely agree with you that it makes more sense to keep the separation.

**Comment by balupton** [comment]
done and working, each new passing master will now do a npm release to the v2.0.x space all automatically

**Comment by balupton** [comment]
upgrading the docpad website now

**Comment by Manvel** [comment]
@balupton out of curiosity - How did you solve the issue with the Github API hitting Rate limit without leaking API credentials to public Travis CI setup?

**Comment by balupton** [comment]
all done

> @balupton out of curiosity - How did you solve the issue with the Github API hitting Rate limit without leaking API access credentials to public Travis CI setup?

- https://github.com/bevry/githubauthquerystring
- https://github.com/bevry/getrepos/blob/master/HISTORY.md#v130-2018-november-28
- then use `travis env set key value` for the credentials

---

for your version of it, send a PR probably with a fresh branch to the repo — however, for now, I am off to bed — also open to other suggestions for how to do it better

for now, https://github.com/bevry/staticsitegenerators-website/issues/30 seems out best bet to solve the update issue
